src/controllers/category.controller.ts

```diff
@@ -1,76 +1,122 @@
-            }
-            return res.status(500).json({ message: "Internal server error" });
-        }
-    }
+  if (trimmedName === 0) {
+    return next(new ApplicationError(400, "Category Name is required"));
+  }
 
-    async getCategoryById(req: Request, res: Response) {
-        try {
-            const { id } = req.params;
-            const category = await categoryService.getCategoryById(id);
-            return res.status(200).json(category);
-        } catch (error) {
-            if (error instanceof ApplicationError) {
-                return res.status(error.statusCode).json({ message: error.message });
-            }
-            return res.status(500).json({ message: "Internal server error" });
-        }
-    }
+  try {
+    const existingCategory = await categoryRepository.findOne({
+      where: { name },
+    });
 
-    async updateCategory(req: Request, res: Response) {
-        try {
-            const { id } = req.params;
-            const categoryData = req.body;
-            const updatedCategory = await categoryService.updateCategory(id, categoryData);
-            return res.status(200).json(updatedCategory);
-        } catch (error) {
-            if (error instanceof ApplicationError) {
-                return res.status(error.statusCode).json({ message: error.message });
-            }
-            return res.status(500).json({ message: "Internal server error" });
-        }
+    if (existingCategory) {
+      return res
+        .status(400)
+        .json({ message: "Category with this name already exists." });
     }
 
-    async deleteCategory(req: Request, res: Response) {
-        try {
-            const { id } = req.params;
-            await categoryService.deleteCategory(id);
-            return res.status(204).send();
-        } catch (error) {
-            if (error instanceof ApplicationError) {
-                return res.status(error.statusCode).json({ message: error.message });
-            }
-            return res.status(500).json({ message: "Internal server error" });
+    const newCategory = categoryRepository.create({ name, description });
+    const savedCategory = await categoryRepository.save(newCategory);
+
+    return res.status(201).json(savedCategory);
+  } catch (error: any) {
+    console.error("Error creating category:", error);
+    return next(new ApplicationError(500, "Error creating category"));
+  }
+};
+
+export const getAllCategories = async (
+  req: Request,
+  res: Response,
+  next: Function
+) => {
+  try {
+    const categories = await categoryRepository.find();
+    return res.status(200).json(categories);
+  } catch (error: any) {
+    console.error("Error fetching categories:", error);
+    return next(new ApplicationError(500, "Error fetching categories")); // Handle any errors
+  }
+};
+
+export const getCategoryById = async(
+    req: Request,
+    res:Response,
+    next:Function
```

Comment:
Why added next function here ? Is this middleware function ?

Reply:
yes sir because we using error handling middleware